File: TP1/Q16/PalindromoRec.java
import java.util.Scanner;

public class PalindromoRec{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String linha = sc.nextLine();
        while(!(linha.length() == 3 && linha.charAt(0) == 'F' && linha.charAt(1) == 'I' && linha.charAt(2) == 'M' )){
            ehPalindromo(linha);
            linha= sc.nextLine();
        }

    }

    public static void ehPalindromo(String linha){
        boolean ehPalindromo = true;
        ehPalindromo=analisando(linha,0, linha.length()/2, linha.length()-1);
        if(ehPalindromo){
            System.out.println("SIM");
        }else{
            System.out.println("NAO");
        }
    }

    public static boolean analisando(String linha, int i, int n, int atras){
        boolean ehPalindromo= true;
        if(i<n && i < atras){
            if(linha.charAt(i) != linha.charAt(atras)){
                ehPalindromo = false;
                i= n;
            }
            if (ehPalindromo) {
                ehPalindromo = analisando(linha, i+1, n, atras-1);
            }
        } 

        return ehPalindromo;
    }

}

File: TP1/Q17/PalindromoRec.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

void converterASC(int strNum[], char linha[], int i){
    if( i < strlen(linha)){

        int numASC = (int)linha[i];
        strNum[i]= numASC;
        converterASC(strNum, linha, i+1);
    }

}

bool teste(int strNum[], int i, int atras){
    bool ehPalindromo = true;
    if( strNum[i] != strNum[atras]){
        ehPalindromo=false;
        i= atras;
    }
    if(i< atras){
        teste(strNum, i+1, atras-1);
    }

    return ehPalindromo;
}

int main(){
    
    char linha[200];
    fgets(linha,200,stdin);
    linha[strcspn(linha,"\n")] = '\0';
    while(!(strlen(linha) == 3 && linha[0] == 'F' && linha[1] == 'I' && linha[2] == 'M')){

        int strNum[strlen(linha)];
        converterASC(strNum, linha, 0);
        bool ehPalindromo = teste(strNum, 0, strlen(linha)-1);
        if(ehPalindromo){
            printf("SIM\n");
        }else{
            printf("NAO\n");
        }
        fgets(linha, 200, stdin);
        linha[strcspn(linha, "\n")] = '\0';
    }

    return 0;
}
